articles/2020-04-25-partager-donnees-sensible-git-chiffrement.md

```diff
@@ -0,0 +1,167 @@
+---
+date: 2020-04-25
+title: Partager des données sensibles de façon sécurisé grâce à Git et au chiffrement
+author: Aschen
+slug: partager-donnees-sensibles-securise-git-chiffrement
+---
+
+_[English version available on Kuzzle blog](https://blog.kuzzle.io/share-sensitive-data-with-git-and-cryptography)_
+
+Tous les projets ont besoin d’un certain nombre de données qui doivent **rester secrètes et privées**.
+
+On parle ici des **clé d’API externes**, des **identifiants à des services tiers**, de **certificats** ou encore de **clés privés SSH**.
+
+Le vrai problème c'est que plus un projet évolue et plus le nombre de secrets à partager pour le bon fonctionnement de l'application augmente
+
+**TLDR;**
+ - N’utilisez **pas de variables d’environnement** pour partager vos clés d’API et autres secrets, à la place utilisez un **fichier JSON chiffré**.
+ - Utilisez [Kourou](https://github.com/kuzzleio/kourou) pour **chiffrer/déchiffrer** les fichiers contenant vos secrets 
+ - Utilisez [Kuzzle Vault](https://github.com/kuzzleio/kuzzle-vault) pour **utiliser les secrets déchiffrés** dans votre application
+
+## Avant on avait des variables d’environnement
+
+La manière la plus simple et la plus répandue de partager des secrets est via les **variables d’environnement**.
+
+On peut rapidement se trouver avec **une dizaine de clés d’API différentes**. Multiplié par le nombre d’environnement de déploiement cible on se retrouve rapidement avec beaucoup trop de variables d’environnements à gérer.
+
+Le problème c’est qu’il est nécessaire de **fournir l’ensemble des variables à chaque personne** qui va devoir travailler sur le projet. 
+
+```bash
+$ SENTRY_DSN=********** AWS_S3_KEY_ID=******** AWS_S3_SECRET_KEY=********* \
+SENDGRID_API_KEY=******** TWILIO_API_KEY=******** KUZZLE_VAULT_KEY=******** \
+ABEEWAY_TOKEN=********* WIREPAS_PASSWORD=******* CLOUDINARY_API_KEY=********* \
+STRIPE_TOKEN=******* node server.js
+```
+
+Et on ne parle même pas du boulot nécessaire à nos cher Devops pour maintenir l’ensemble de ces variables pour plusieurs environnements de déploiement (production, staging, development, etc)
```

Comment:
Peut-être ajouter une mention du nombre maximum de variables d'environnements sur certains envs

Reply:
J'arrive pas vraiment à trouver une limite, à priori y'a une limite à la taille total de la string `ENV` mais assez élevé à priori https://stackoverflow.com/questions/1078031/what-is-the-maximum-size-of-an-environment-variable-value